backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -348,17 +358,50 @@ public Response create(@PathParam("vaultId") UUID vaultId, @Valid VaultDto vault
 		}
 	}
 
+	@PATCH
```

Comment:
I deliberately used PATCH, because `PUT` is used for creating a vault:
https://github.com/cryptomator/hub/blob/7aa07c12fad67fdfea04af89639a57d72680d721/backend/src/main/java/org/cryptomator/hub/api/VaultResource.java#L328-L330

According to the specs we cannot put the same vault twice. I guess the reason was that PUT replaces an object, meaning we could also replace (in the code) a masterkey of a vault. Of course, like you mentioned earlier, we can make the backend dumb and just accept what we put. But then we would have the non-zero risk of destroying keymaterial.

Reply:
The server is still free to reject any changes to existing objects that it doesn't like. Or is there any spec that says that a PUT _must_ overwrite existing data, no questions asked?

Reply:
> The server is still free to reject any changes to existing objects that it doesn't like. Or is there any spec that says that a PUT must overwrite existing data, no questions asked?

Of course not, but that would add logic to our backend again, instead of having a dumb one. And how should the server react on fields that should not change? Should those be ignored or a bad request code returned? Logged at least? And we also would drop the 409 code, because there is no distinction possible anymore between "update" and "already exists".

Reply:
After a discussion we have a plan now: The  `/vaults/{vaultid}` PATCH endpoint will be removed again. The update functionality will be merged into the respective PUT endpoint. Done in https://github.com/cryptomator/hub/pull/202/commits/a21369f2be3497e5d8456a3ff9a1761ae1d85358